Package Request for DNSTOP

<!-- Important note: motivating 'why this package is needed' will most likely give the package higher priority in the developers TODO-list -->

**Package description**
dnstop is a libpcap application (like tcpdump) that displays various tables of DNS traffic on your network. Currently dnstop displays tables of:

Source IP addresses
Destination IP addresses
Query types
Response codes
Opcodes
Top level domains
Second level domains
Third level domains
etc...
+<http://dns.measurement-factory.com/tools/dnstop/>


**Link to home page and sources**
Home page:http://dns.measurement-factory.com/tools/dnstop/.

DNSTOP is could be a cool feature in termux to enhance and enable dns registries made available in android instead of your PC. I really need dnstop for project use and also for others use since it is a helpful tool to log dns records of network connectivity. But unfortunately, I tried compiling DNSTOP then there were errors which I am unfamiliar with(*no background yet of C programs*) so could you help me out on this?
Entry logs like "struct " error. I was able to get the necessary dependencies but still couldn't figure out the problem in compiling it. :(
$ make install
gcc -g -O2 -DUSE_IPV6=1   -c -o dnstop.o dnstop.c
dnstop.c:691:35: warning: taking address of packed member 'ip6_src' of
      class or structure 'ip6_hdr' may result in an unaligned pointer
      value [-Waddress-of-packed-member]
    inXaddr_assign_v6(&src_addr, &ipv6->ip6_src);
                                  ^~~~~~~~~~~~~
dnstop.c:692:35: warning: taking address of packed member 'ip6_dst' of
      class or structure 'ip6_hdr' may result in an unaligned pointer
      value [-Waddress-of-packed-member]
    inXaddr_assign_v6(&dst_addr, &ipv6->ip6_dst);
                                  ^~~~~~~~~~~~~
dnstop.c:856:9: error: use of undeclared identifier 'ETHERTYPE_IP'
    if (ETHERTYPE_IP == etype) {
        ^
dnstop.c:866:35: error: incomplete definition of type
      'struct ether_header'
    unsigned short etype = ntohs(e->ether_type);
                                 ~^
/data/data/com.termux/files/usr/include/sys/endian.h:65:27: note:
      expanded from macro 'ntohs'
#define ntohs(x) __swap16(x)
                          ^
dnstop.c:865:12: note: forward declaration of 'struct ether_header'
    struct ether_header *e = (void *)pkt;
           ^
2 warnings and 2 errors generated.
make: *** [<builtin>: dnstop.o] Error 1

--- a/packages/dnstop/build.sh
+++ b/packages/dnstop/build.sh
@@ -1,0 +1,27 @@
+TERMUX_PKG_HOMEPAGE=https://github.com/measurement-factory/dnstop
+TERMUX_PKG_DESCRIPTION="A libpcap application that displays various tables of DNS traffic on your network"
+TERMUX_PKG_LICENSE="BSD 3-Clause"
+TERMUX_PKG_MAINTAINER="@termux"
+_COMMIT=1fc98eb479513f4be68ba500aec18934d24e8fd6
+TERMUX_PKG_VERSION=2021.03.29
+TERMUX_PKG_SRCURL=https://github.com/measurement-factory/dnstop.git
+TERMUX_PKG_GIT_BRANCH=master
+TERMUX_PKG_DEPENDS="libpcap, ncurses"
+TERMUX_PKG_BUILD_IN_SRC=true
+
+termux_step_post_get_source() {
+	git fetch --unshallow
+	git checkout $_COMMIT
+
+	local version="$(git log -1 --format=%cs | sed 's/-/./g')"
+	if [ "$version" != "$TERMUX_PKG_VERSION" ]; then
+		echo -n "ERROR: The specified version \"$TERMUX_PKG_VERSION\""
+		echo " is different from what is expected to be: \"$version\""
+		return 1
+	fi
+	sed -i "s/@VERSION@/$version/g" dnstop.c
+}
+
+termux_step_pre_configure() {
+	CPPFLAGS+=" -D__USE_BSD"
+}

